### DISCUSSION TITLE
Why does `lua_ls`'s recommended config only set settings if `.luarc.json` doesn't exist?

### DISCUSSION THREAD

**Discussion by mawkler** [neovim/neovim#25964] [Q&A]
[lua_ls's recommended configuration](https://github.com/neovim/nvim-lspconfig/blob/b0852218bc5fa6514a71a9da6d5cfa63a263c83d/doc/server_configurations.md?plain=1#L5357C9-L5357C9) is only set if the workspace doesn't have a `.luarc.json` or `.luarc.jsonc`. Why is that?

**Comment by clason** [comment]
You've got to ask nvim-lspconfig for that; it's a separate project.

**Reply by mawkler** [reply]
@clason I was going to, but it says there to [ask to ask quesions here](https://github.com/neovim/nvim-lspconfig#contributions):

> Ask questions on [GitHub Discussions](https://github.com/neovim/neovim/discussions) or in the [Neovim Matrix](https://app.element.io/#/room/#neovim:matrix.org) room.

🙂

**Comment by lewis6991** [comment]
Because the `.luarc.json` should specify the workspace.

**Reply by mawkler** [reply]
@lewis6991 Ok, that makes sense, but I'm not quite following. Doesn't the recommended configuration mean that your personal lua_ls configuration should _not_ apply if a `luarc.json` is found? Why is that the recommended behaviour?

**Reply by lewis6991** [reply]
Settings belong at the project level, not the individual user level.

The fallback settings are just for when there is no project settings.

You do you though.
